### ISSUE TITLE
Emtpy string as directory separator?

### ISSUE THREAD

**Issue by markusschaber** [dotnet/docs#35431]
The "Remarks" Section says:

> Directory separators "/" and "" are allowed in hintName, they are normalized to "/" regardless of host platform.

As far as I know, the empty string is not really a directory separator. I guess this ought to be a backslash, and has been lost in quoting and escaping mechanisms somewhere.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 8eaaac31-2551-8da3-ff44-a7770fed9fa3
* Version Independent ID: ae740f38-097d-9f66-5e76-9dcfb3bc5066
* Content: [GeneratorExecutionContext.AddSource Method (Microsoft.CodeAnalysis)](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.generatorexecutioncontext.addsource?view=roslyn-dotnet-4.6.0)
* Content Source: [dotnet/xml/Microsoft.CodeAnalysis/GeneratorExecutionContext.xml](https://github.com/dotnet/roslyn-api-docs/blob/live/dotnet/xml/Microsoft.CodeAnalysis/GeneratorExecutionContext.xml)
* Product: **dotnet-roslyn-api**
* Technology: **microsoft.codeanalysis**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by peteraritchie** [comment]
Is this open-source somewhere? It's been 3 years, maybe someone from the community can do a PR for this?

**Comment by peteraritchie** [comment]
> Is this open-source somewhere? It's been 3 years, maybe someone from the community can do a PR for this?

Found it: https://github.com/dotnet/roslyn/blob/80fa496bcade32450dc3fc82b43607d0df0c5631/src/Compilers/Core/Portable/SourceGeneration/IncrementalContexts.cs#L213

**Comment by CyrusNajmabadi** [comment]
Agreed.  This is fine.  If there's a docs issue, it would be in post processing of this.
